apps/cowswap-frontend/src/modules/swap/hooks/useSwapConfirmManager.ts

```diff
@@ -1,75 +1,110 @@
-import { useAtom } from 'jotai'
+import { useSetAtom } from 'jotai'
 import { useMemo } from 'react'
 
 import TradeGp from 'legacy/state/swap/TradeGp'
 import { useExpertModeManager } from 'legacy/state/user/hooks'
 
-import { swapConfirmAtom } from 'modules/swap/state/swapConfirmAtom'
+import { swapConfirmAtom, SwapConfirmState } from 'modules/swap/state/swapConfirmAtom'
 
 export interface SwapConfirmManager {
   setSwapError(swapErrorMessage: string): void
-  openSwapConfirmModal(tradeToConfirm: TradeGp): void
+  openSwapConfirmModal(tradeToConfirm: TradeGp, needsPermitSignature?: boolean): void
   acceptRateUpdates(tradeToConfirm: TradeGp): void
   closeSwapConfirm(): void
   sendTransaction(tradeToConfirm: TradeGp): void
   transactionSent(txHash: string): void
+  requestPermitSignature(): void
+  permitSigned(): void
 }
 
 export function useSwapConfirmManager(): SwapConfirmManager {
-  const [swapConfirmState, setSwapConfirmState] = useAtom(swapConfirmAtom)
+  const setSwapConfirmState = useSetAtom(swapConfirmAtom)
```

Comment:
Simplified this part turning into a setter only.
We don't need to read the value anywhere, and actually depending on that to update the state causes race conditions.
Replaced (where needed) with the setter taking a fn instead of the new state